src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AzureConnection.h"
+#include "AzureClientID.h"
+#include <sstream>
+#include <stdlib.h>
+
+// STARTF_USESTDHANDLES is only defined in WINAPI_PARTITION_DESKTOP
+// We're just gonna manually define it for this prototyping code
+#ifndef STARTF_USESTDHANDLES
+#define STARTF_USESTDHANDLES 0x00000100
+#endif
+
+#include "AzureConnection.g.cpp"
+
+#include "../../types/inc/Utils.hpp"
+
+using namespace ::Microsoft::Console;
+
+using namespace utility;
+using namespace web;
+using namespace web::json;
+using namespace web::http;
+using namespace web::http::client;
+using namespace web::websockets::client;
+using namespace concurrency::streams;
+
+namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+{
+    AzureConnection::AzureConnection(const uint32_t initialRows, const uint32_t initialCols) :
+        _initialRows{ initialRows },
+        _initialCols{ initialCols },
+        _loginUri{ U("https://login.microsoftonline.com/") },
+        _resourceUri{ U("https://management.azure.com/") },
+        _wantedResource{ U("https://management.core.windows.net/") }
+    {
+    }
+
+    winrt::event_token AzureConnection::TerminalOutput(Microsoft::Terminal::TerminalConnection::TerminalOutputEventArgs const& handler)
+    {
+        return _outputHandlers.add(handler);
+    }
+
+    void AzureConnection::TerminalOutput(winrt::event_token const& token) noexcept
+    {
+        _outputHandlers.remove(token);
+    }
+
+    winrt::event_token AzureConnection::TerminalDisconnected(Microsoft::Terminal::TerminalConnection::TerminalDisconnectedEventArgs const& handler)
+    {
+        return _disconnectHandlers.add(handler);
+    }
+
+    void AzureConnection::TerminalDisconnected(winrt::event_token const& token) noexcept
+    {
+        _disconnectHandlers.remove(token);
+    }
+
+    void AzureConnection::Start()
+    {
+        // Create our own output handling thread
+        // Each connection needs to make sure to drain the output from its backing host.
+        _hOutputThread.reset(CreateThread(nullptr,
+                                          0,
+                                          StaticOutputThreadProc,
+                                          this,
+                                          0,
+                                          nullptr));
+
+        _connected = true;

```

Comment:
You're not checking if `CreateThread` actually succeeded.

If `ConhostConnection` isn't doing that either, we have two bugs.